### PR TITLE
[IMP] stock_request_picking_type: Set Picking Type in Create

### PR DESCRIPTION
**Pull request by osi-scampbell**
Added method to set the Picking Type in the create() based on the warehouse_id

### CODE REVIEW

File: stock_request_picking_type/models/stock_request_order.py [hunk 1]
```diff
@@ -28,3 +28,14 @@ def onchange_warehouse_picking_id(self):
...
+
+
+    @api.model
+    def create(self, vals):
+        if vals.get('warehouse_id', False):
+            picking_type_id = self.env['stock.picking.type'].\
+                search([('code', '=', 'stock_request_order'),
+                        ('warehouse_id', '=', vals['warehouse_id'])], limit=1)
+            if picking_type_id:
+                vals['picking_type_id'] = picking_type_id.id
```
Comment (max3903) [inline stock_request_picking_type/models/stock_request_order.py]:
```suggestion
                vals.update({'picking_type_id': picking_type_id.id})
```

**Review by max3903 — APPROVED** [review]

**Review by patrickrwilson — APPROVED** [review]
Functional Test confirmed it works.

File: stock_request_picking_type/models/stock_request_order.py [hunk 2]
```diff
@@ -28,3 +28,14 @@ def onchange_warehouse_picking_id(self):
                         ('warehouse_id', '=', self.warehouse_id.id)], limit=1)
             if picking_type_id:
                     self._origin.write({'picking_type_id': picking_type_id.id})
+
```
Comment (max3903) [inline stock_request_picking_type/models/stock_request_order.py:31]:
```suggestion
```

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**